airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeS3StreamCopier.java

```diff
@@ -66,24 +78,52 @@ public SnowflakeS3StreamCopier(final String stagingFolder,
         MAX_PARTS_PER_FILE);
   }
 
+  @Override
+  public void copyStagingFileToTemporaryTable() throws Exception {
```

Comment:
is this completely identical to the methods in the GCS copier? if yes, let's extract them to a shared implementation rather than copy-pasting

Reply:
@edgao extracted shared implementation into SnowflakeParallelCopyStreamCopier interface